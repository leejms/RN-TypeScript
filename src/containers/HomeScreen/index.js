import React from 'react'
import { Button, ScrollView } from 'react-native'

import CenteredTitle from '../../components/text/CenteredTitle'

const HomeScreen = ({navigation}) => {
  return (
    <ScrollView>
      <CenteredTitle title='Hello Home!' />
      <Button
        onPress={() => navigation.navigate('Profile', { name: 'Jane' })}
        title='Go to a profile screen'
      />
    </ScrollView>
  )
}

HomeScreen.navigationOptions = {
  title: 'Welcome'
}

export default HomeScreen
