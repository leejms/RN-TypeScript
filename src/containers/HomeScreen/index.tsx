import React from 'react'
import { Button, ScrollView } from 'react-native'

import CenteredTitle from '../../components/text/CenteredTitle'

// tslint:disable-next-line:variable-name
const HomeScreen = ({navigation}) => {
  return (
    <ScrollView>
      <CenteredTitle title='Hello Home!' />
      <Button
        onPress={() => navigation.navigate('Profile', { name: 'Jane' })}
        title='Go to a profile screen'
      />
    </ScrollView>
  )
}

export default HomeScreen
