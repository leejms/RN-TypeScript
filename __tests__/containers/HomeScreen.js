import 'react-native'
import React from 'react'
import HomeScreen from '../../src/containers/HomeScreen'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('<HomeScreen />', () => {
  it ('matches snapshot', () => {
    const tree = renderer.create(
      <HomeScreen />
    ).toJSON()
    expect (tree).toMatchSnapshot()
  })
})
