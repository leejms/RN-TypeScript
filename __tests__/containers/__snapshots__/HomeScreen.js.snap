// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> matches snapshot 1`] = `
<RCTScrollView>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "borderRadius": 4,
          "borderWidth": 0,
          "fontSize": 24,
          "paddingTop": 20,
          "textAlign": "center",
        }
      }
    >
      Hello Home!
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Go to a profile screen
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Go back
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
